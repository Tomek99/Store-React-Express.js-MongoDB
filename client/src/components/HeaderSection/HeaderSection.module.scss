.headerSection {
    text-align: center;
    font-size: 5rem;
    font-weight: bold;
    text-transform: uppercase;
    margin-bottom: 20px;

    span:nth-of-type(1) {
        color: #fff;
    }
    span:nth-of-type(2) {
        color: var(--main-color);
    }
}

@media screen and (max-width: 800px) {
    .headerSection {
        font-size: 4.5rem;
    }
}

@media screen and (max-width: 400px) {
    .headerSection {
        font-size: 3.5rem;
    }
}
