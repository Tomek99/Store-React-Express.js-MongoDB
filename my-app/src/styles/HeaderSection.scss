div.divHeader {
    position: sticky;
    top:0;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    background-color: var(--black);
    border-bottom: var(--border);

    div.logo {
        cursor: pointer;
    }

    ul.navBar {
        display: flex;
        flex-direction: row;
        list-style-type: none;
        gap: 20px;
        font-size: 2rem;

        li > a {
            position: relative;
            color: #fff;
        }
        a:hover {
            color: var(--main-color);
        }

        a:hover::after{
           display: block;
           content: '';
           width: 100%;
           height: 1px;
           background: var(--main-color);
           position: absolute;
           bottom: 0;
           left: 0;
        }

    }

    div.buttonsNextToEO {
        display: flex;
        gap: 10px;
        
        button {
            background-color: transparent;
            cursor: pointer;
        }
    }
}