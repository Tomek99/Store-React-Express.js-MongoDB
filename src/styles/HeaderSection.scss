.HeaderSection {
    position: sticky;
    top: 0;
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: var(--black);
    border-bottom: var(--border);
    z-index: 99;

    .hamburger {
        display: none;
        cursor: pointer;
        position: absolute;
        top: 15px;
        left: 15px;

        .bar {
            display: block;
            width: 25px;
            height: 3px;
            margin: 5px auto;
            -webkit-transition: all 0.3s ease-in-out;
            transition: all 0.3s ease-in-out;
            background-color: #fff;
        }
    }

    .navBar {
        max-width: 1620px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: space-around;

        div {
            cursor: pointer;

            img.logo {
                width: 100%;
                margin: 5px 0 5px 0;
            }
        }

        .navMenu {
            display: flex;
            flex-direction: row;
            align-items: center;
            list-style-type: none;
            gap: 20px;
            font-size: 2rem;

            li > a {
                position: relative;
                color: #fff;
            }
            a:hover {
                color: var(--main-color);
            }

            a:hover::after {
                display: block;
                content: "";
                width: 100%;
                height: 1px;
                background: var(--main-color);
                position: absolute;
                bottom: 0;
                left: 0;
            }
        }

        .btn-section {
            display: flex;
            position: relative;
            gap: 10px;

            button {
                background-color: transparent;
                cursor: pointer;
            }
        }
    }
}

@media screen and (max-width: 900px) {
    .hamburger {
        display: block !important;
    }

    .hamburger.active .bar:nth-child(2) {
        opacity: 0;
    }
    .hamburger.active .bar:nth-child(1) {
        transform: translateY(8px) rotate(45deg);
    }
    .hamburger.active .bar:nth-child(3) {
        transform: translateY(-8px) rotate(-45deg);
    }

    .navMenu {
        position: fixed;
        left: -100%;
        top: 76px;
        flex-direction: column !important;
        background-color: var(--bg);
        width: 40%;
        text-align: center;
        transition: 0.3s;
        border-bottom: var(--border);
        border-right: var(--border);

        .nav-item {
            margin: 16px;
        }
    }

    .navMenu.active {
        left: 0;
    }

    img.logo {
        width: 80% !important;
    }

    .PopupBasket {
        top: 76px !important;
    }
}
