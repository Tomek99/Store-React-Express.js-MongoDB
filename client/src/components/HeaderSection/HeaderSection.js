import React from "react";
import styles from "./HeaderSection.module.scss";
import PropTypes from "prop-types";

function Header({ firstWord, secondWord }) {
  return (
    <header className={styles.headerSection}>
      {firstWord} {secondWord}
    </header>
  );
}

Header.propTypes = {
  firstWord: PropTypes.string,
  secondWord: PropTypes.string,
};

export default Header;
