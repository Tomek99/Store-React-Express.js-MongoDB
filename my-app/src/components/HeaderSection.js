import React, { useRef, useState } from 'react'
import '../styles/HeaderSection.scss'
import Logo from '../images/logo.png'
import { BsCartFill, BsSearch } from 'react-icons/bs'
import PopupSearch from './PopupSearch';


function Header() {
    const [isOpen, setIsOpen] = useState(false);
    const [isMenuOpen, setMenuOpen] = useState(false);

    const handleSearch = () => {
        setIsOpen(!isOpen);
    }

    const unfoldMenu = () => {
        setMenuOpen(!isMenuOpen);
        console.log(isMenuOpen);
    }


    return (
        <>
            <div className='divHeader'>
                <div className='menuHamburger'><button type='button' onClick={unfoldMenu}>X</button></div>
                <div className='bar' >
                    <div className='logo'>
                        <a href="#homeSection"><img src={Logo} alt="" /></a>
                    </div>
                    <div className="hideBar" style={{ display: isMenuOpen ? 'none' : 'block' }}>
                        <ul className='navBar'>
                            <li><a href='#homeSection'>Home</a></li>
                            <li><a href='#aboutSection'>About</a></li>
                            <li><a href='#menuSection'>Menu</a></li>
                            <li><a href='#productsSection'>Products</a></li>
                            <li><a href='#reviewSection'>Review</a></li>
                            <li><a href='#contactSection'>Contact</a></li>
                            <li><a href='#blogsSection'>Blogs</a></li>
                        </ul>
                    </div>

                    <div className='buttonsNextToEO' style={{ display: isMenuOpen ? 'none' : 'block' }}>

                        <button type="button" onClick={handleSearch}><BsSearch size={30} color={"#fff"} onMouseOver={({ target }) => target.style.color = "d3ad7f"} onMouseOut={({ target }) => target.style.color = "#fff"} /></button>

                        <button type="button"><BsCartFill size={30} color={"#fff"} onMouseOver={({ target }) => target.style.color = "d3ad7f"} onMouseOut={({ target }) => target.style.color = "#fff"} /></button>
                    </div>
                </div>
            </div>

        </>
    )
}

export default Header