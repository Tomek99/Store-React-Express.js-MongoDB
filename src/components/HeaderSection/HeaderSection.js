import React, { useRef, useState } from 'react'
import '../../styles/HeaderSection.scss'
import { BsCartFill, BsSearch } from 'react-icons/bs'
import PopupSearch from '../PopupSearch/PopupSearch';
import PopupBasket from '../PopupBasket/PopupBasket'

function Header() {
    const [isBasketOpen, setBasketOpen] = useState(false);
    const [isNavigationOpen, setNavigationOpen] = useState(false);
    let setClass = isNavigationOpen ? 'navMenu active' : 'navMenu';
    // const handleSearch = () => {
    //     setIsOpen(!isOpen);
    // }

    const handleNavigation = () => {
        setNavigationOpen(!isNavigationOpen);
        setBasketOpen(false);
    }

    const handleBasket = () => {
        setBasketOpen(!isBasketOpen);
        setNavigationOpen(false);
        console.log("test")
    }

    return (
        <>
            <div className='HeaderSection'>
                <div className={isNavigationOpen ? 'hamburger active' : 'hamburger'} onClick={handleNavigation}>
                    <span className='bar'></span>
                    <span className='bar'></span>
                    <span className='bar'></span>
                </div>
                <div className='navBar' >
                    <div className='logo'>
                        <a href="#homeSection"><img src="/images/logo.png" alt="" /></a>
                    </div>
                    <nav >
                        <ul className={setClass} >
                            <li className='nav-item'><a href='#homeSection' className='nav-link'>Home</a></li>
                            <li className='nav-item'><a href='#aboutSection' className='nav-link'>About</a></li>
                            <li className='nav-item'><a href='#menuSection' className='nav-link'>Menu</a></li>
                            <li className='nav-item'><a href='#productsSection' className='nav-link'>Products</a></li>
                            <li className='nav-item'><a href='#reviewSection' className='nav-link'>Review</a></li>
                            <li className='nav-item'><a href='#contactSection' className='nav-link'>Contact</a></li>
                            <li className='nav-item'><a href='#blogsSection' className='nav-link'>Blogs</a></li>
                        </ul>
                    </nav>
                    <div className='buttonsNextToEO'>

                        <button type="button" ><BsSearch size={30} color={"#fff"} onMouseOver={({ target }) => target.style.color = "d3ad7f"} onMouseOut={({ target }) => target.style.color = "#fff"} /></button>

                        <button type="button" onClick={handleBasket}><BsCartFill size={30} color={"#fff"} onMouseOver={({ target }) => target.style.color = "d3ad7f"} onMouseOut={({ target }) => target.style.color = "#fff"} /></button>
                        <PopupBasket isBasketOpen={isBasketOpen} />
                    </div>
                </div>
            </div>

        </>
    )
}

export default Header