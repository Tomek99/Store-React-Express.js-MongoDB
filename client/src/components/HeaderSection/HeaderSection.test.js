import { render } from "@testing-library/react";
import HeaderSection from "./HeaderSection";

test("should render HeaderSection component", () => {
  const { container } = render(
    <HeaderSection firstWord="Our" secondWord="products" />
  );
  expect(container).toMatchSnapshot();
});
