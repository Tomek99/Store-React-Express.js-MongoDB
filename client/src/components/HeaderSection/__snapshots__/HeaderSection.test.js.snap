// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeaderSection component 1`] = `
<div>
  <header
    class="headerSection"
  >
    <span>
      Our
    </span>
     
    <span>
      products
    </span>
  </header>
</div>
`;
